src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutValenceCheck.java

```diff
@@ -0,0 +1,167 @@
+    {
+        final Edge edge = (Edge) object;
+
+        // Get all edges in the roundabout
+        final Set<Edge> roundaboutEdges = getAllRoundaboutEdges(edge);
+
+        final Set<Edge> connectedEdges = roundaboutEdges.stream()
+                .flatMap(roundaboutEdge -> roundaboutEdge.connectedEdges().stream())
+                .filter(HighwayTag::isCarNavigableHighway).filter(Edge::isMasterEdge)
+                .filter(e -> !JunctionTag.isRoundabout(e)).filter(e -> !roundaboutEdges.contains(e))
+                .collect(Collectors.toSet());
+        final int totalRoundaboutValence = connectedEdges.size();
+
+        // If the totalRoundaboutValence is less than the minimum configured number of connections
+        // or greater than or equal to the maximum configured number of connections
+        if (totalRoundaboutValence < this.minimumValence
+                || totalRoundaboutValence >= this.maximumValence)
+        {
+            this.markAsFlagged(object.getIdentifier());
+
+            // If the roundabout valence is 1, this should be labelled as a turning loop instead
+            if (totalRoundaboutValence == 1)
+            {
+                return Optional.of(this.createFlag(roundaboutEdges,
+                        this.getLocalizedInstruction(1, edge.getOsmIdentifier())));
+            }
+            // Otherwise, we want to flag and given information about identifier and valence
+            return Optional.of(this.createFlag(roundaboutEdges, this.getLocalizedInstruction(0,
+                    edge.getOsmIdentifier(), totalRoundaboutValence)));
+        }
+        // If the totalRoundaboutValence is not unusual, we don't flag the object
+        else
+        {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * This method gets all edges in a roundabout given one edge in that roundabout
+     * 
+     * @param edge
+     *            An Edge object known to be a roundabout edge
+     * @return A set of edges in the roundabout
+     */
+    private Set<Edge> getAllRoundaboutEdges(final Edge edge)
+    {
+        final Set<Edge> roundaboutEdges = new HashSet<>();
+        // Initialize a queue to add yet to be processed connected edges to
+        final Queue<Edge> queue = new LinkedList<>();
+
+        // Mark the current node as visited and enqueue it
+        this.markAsFlagged(edge.getIdentifier());
+        queue.add(edge);
+
+        // As long as the queue is not empty
+        while (queue.size() != 0)
+        {
+            // Dequeue a connected edge and add it to the roundaboutEdges
+            final Edge e = queue.poll();
+
+            // Check to make sure that we are only adding master edges into the set
+            if (!e.isMasterEdge())
+            {
+                continue;
+            }
+
+            roundaboutEdges.add(e);
+
+            // Get the edges connected to the edge e as an iterator
+            final Iterator<Edge> iterator = e.connectedEdges().iterator();
```

Comment:
I'd suggest using a for loop unless there is an advantage of using iterators here. `connectedEdges` method gives you a set to iterator over right away.